misk/src/main/kotlin/misk/security/csp/CspInterceptor.kt

```diff
@@ -0,0 +1,22 @@
+package misk.security.csp
+
+import misk.Action
+import misk.web.NetworkChain
+import misk.web.NetworkInterceptor
+import javax.inject.Inject
+import kotlin.reflect.full.findAnnotation
+
+class CspInterceptor(val rules: List<String>) : NetworkInterceptor {
+  override fun intercept(chain: NetworkChain) {
+    chain.httpCall.setResponseHeader("Content-Security-Policy", rules.joinToString(separator = "; ", postfix = ";"))
```

Comment:
Thought for a follow up PR - should we also validate the syntax and grammar of the policy?

Reply:
Yes, but not yet.